src/cli.ts

```diff
@@ -1,13 +1,286 @@
+    console.log(`${targetDir} exists and is not empty. No restructuring needed.`);
+    return;
+  }
+  console.log(`Starting restructure of ${folders.length} directories`)
+  mkdirSync(targetDir, {recursive: true});
+  for (let folder of folders){
+    console.log(`Copying ${folder} to ${targetDir}/${basename(folder)}`)
+    cpSync(folder, `${targetDir}/${basename(folder)}`, {recursive: true});
+  }
+  console.log(`Sucsessfully restructured ${folders.length} directories`)
+}
+
+async function restructureIfNeeded(rootDir: string) {
+  // before
+  // $rootdir/My Album 1
+  // $rootdir/My Album 2
+  // $rootdir/Photos from 2008
+
+  // after
+  // $rootdir/Albums/My Album 1
+  // $rootdir/Albums/My Album 2
+  // $rootdir/Photos/Photos from 2008
+
+  const photosDir: string = `${rootDir}/Photos`;
+
+  // move the "Photos from $YEAR" directories to Photos/
+  _restructureIfNeeded(await glob(`${rootDir}/Photos from */`), photosDir);
+  
+  // move everythingg else to Albums/, so we end up with two top level folders
+  const fullSet: Set<string> = new Set(await glob(`${rootDir}/*/`));
+  const photoSet: Set<string> = new Set(await glob(`${rootDir}/Photos from */`));
+  photoSet.add(`${rootDir}/Photos`);
+  const everythingExceptPhotosDir: string[] = Array.from(
+    new Set([...fullSet].filter((x) => !photoSet.has(x)))
+  );
+  _restructureIfNeeded(everythingExceptPhotosDir, `${rootDir}/Albums`);
+}
+
+async function runFullMigration(
+  rootDir: string,
+  timeout: number,
+) {
+  // at least in my takeout, the Takeout folder contains a subfolder
+  // Takeout/Google Foto
+  // rootdir refers to that subfolder
+
+  rootDir = (await glob(`${rootDir}/Google*`))[0].replace(/\/+$/, '');
+  await restructureIfNeeded(rootDir);
+  await processPhotos(rootDir, timeout);
+  await processAlbums(rootDir, timeout);
+}
+
+const rewriteAllTags = command({
+  name: 'rewrite all tags for single file',
+  args: {
+    inFile: positional({
+      type: string,
+      displayName: 'in_file',
+      description: 'The path to your input file.',
+    }),
+    outFile: positional({
+      type: string,
+      displayName: 'out_file',
+      description: 'The path to your output location for the file.',
+    }),
+    timeout: option({
+      type: number,
+      defaultValue: () => 30000,
+      short: 't',
+      long: 'timeout',
+      description:
+        'Sets the task timeout in milliseconds that will be passed to ExifTool.',
+    }),
+  },
+  handler: async ({inFile, outFile, timeout}) => {
+    const exifTool = new ExifTool({ taskTimeoutMillis: timeout });
+    await exifTool.rewriteAllTags(inFile, outFile);
+    exifTool.end();
+  },
+});
+
+const fullMigrate = command({
+  name: 'google-photos-migrate-full',
+  args: {
+    takeoutDir: positional({
+      type: string,
+      displayName: 'takeout_dir',
+      description: 'The path to your "Takeout" directory.',
+    }),
+    timeout: option({
```

Comment:
Not sure if it's a ts thing, but this command _only_ works if you run it with the `--timeout ...` after the positional.  That's very non-standard, usually positionals are expected to come after all options.

Reply:
I see it's this same way in the existing `folerMigrate` code though, so I guess just ignore me.

Reply:
It shouldn't need it? It has a default value and all:
```ts
timeout: option({
      type: number,
      defaultValue: () => 30000,
      short: 't',
      long: 'timeout',
      description:
        'Sets the task timeout in milliseconds that will be passed to ExifTool.',
    }),```

Reply:
I only set it because of the README.

Normally from the CLI though, the expectation is that it's  
```
cmd [global-opts] [subcmd [subcmd-opts] [sub-sub-cmd ... ]]] arguments
```

What I was seeing was that I couldn't call it like  
```
$cmd fullMigrate --timeout 60000 /path/to/takeout
```
Which would be standard, I had to call it like  
```
$cmd fullMigrate /path/to/takeout --timeout 60000 
```

Reply:
Yeah that's definitely weird. Unfortunately I think that's just a result of using `cmd-ts` (the CLI library).